### ISSUE TITLE
Date selection is buggy in multiple months shown

### ISSUE THREAD

**Issue by os-hriday-modi** [Hacker0x01/react-datepicker#2807]
Date/Daterange selection is buggy in multiple months shown. When you select a single date for the current month let's say 4th of March 2021, then all the months after March 2021 has 4th as selected date. This gets worse if you have provided `showPreviousMonths` as `true`.

1. Go to 'https://reactdatepicker.com/#example-multiple-months'
2. Click on 'Date Control'
3. Choose a Date of Current Month
4. The chosen date will be selected for all subsequent months.

**Screenshots**

https://user-images.githubusercontent.com/50872244/110819015-cae9aa00-82b3-11eb-9960-1f5a2a5098f8.mov

**Desktop**

- OS: Mac
- Browser: Chrome
- Version 88.0.4324.192

**Comment by matt-tingen** [comment]
I found this confusing as well, but I'm not sure it's a bug per se. The "selection" you're seeing in other months comes from the `.react-datepicker__day--keyboard-selected` class rather than `.react-datepicker__day--selected` . The latter is the actual selection and the former is the cursor for users navigating with the keyboard.

Although this seems to be intended behavior, the default styles for `.react-datepicker__day--keyboard-selected` could possibly be improved. The [current styles](https://github.com/Hacker0x01/react-datepicker/blob/master/src/stylesheets/datepicker.scss#L440) always applies a background which is very close to the actual selection which creates ambiguity and confusion.

In my project, I applied the following styles on top of the defaults. In addition to avoiding the confusing background, it also only shows the indicator when the user is navigating with keyboard.

```css
.react-datepicker__day--keyboard-selected {
  background: inherit;
}
.react-datepicker__day--keyboard-selected:focus {
  outline: 1px dotted black;
}
```

<img width="254" alt="Screen Shot 2021-03-15 at 12 46 08 AM" src="https://user-images.githubusercontent.com/8584728/111104868-fecd0580-8527-11eb-9f96-811034cbd0bf.png">
